src/main/java/com/hms_networks/americas/sc/extensions/datapoint/DataPointNumber.java

```diff
@@ -0,0 +1,148 @@
+package com.hms_networks.americas.sc.extensions.datapoint;
+
+/**
+ * {@link Number} data point class
+ *
+ * <p>Class object for a DataPoint with a {@link Number} value
+ *
+ * @since 1.15.14
+ * @author HMS Networks; Americas
+ */
+public class DataPointNumber extends DataPoint {
+
+  /** Data point value */
+  protected Number value;
+
+  /**
+   * Constructor for a {@link Number} data point with specified quality.
+   *
+   * @param tagName data point tag name
+   * @param tagId data point tag ID
+   * @param tagUnit data point tag Unit
+   * @param value data point value
+   * @param time data point timestamp
+   * @param quality data point quality
+   */
+  public DataPointNumber(
+      String tagName, int tagId, String tagUnit, Number value, String time, DataQuality quality) {
+    this.tagName = tagName;
+    this.tagId = tagId;
+    this.tagUnit = tagUnit;
+    this.value = value;
+    this.timestamp = time;
+    this.quality = quality;
+  }
+
+  /**
+   * Constructor for a {@link Number} data point without specified quality.
+   *
+   * @param tagName data point tag name
+   * @param tagId data point tag ID
+   * @param tagUnit data point tag Unit
+   * @param value data point value
+   * @param time data point timestamp
+   */
+  public DataPointNumber(String tagName, int tagId, String tagUnit, Number value, String time) {
+    this.tagName = tagName;
+    this.tagId = tagId;
+    this.tagUnit = tagUnit;
+    this.value = value;
+    this.timestamp = time;
+    this.quality = DataQuality.GOOD;
+  }
+
+  /**
+   * Get and return the data point value.
+   *
+   * @return data point value
+   */
+  public Number getValue() {
+    return value;
+  }
+
+  /**
+   * Set the data point value.
+   *
+   * <p>This method is unique to the {@link DataPointNumber} class, and is not available in other
+   * implementations of {@link DataPoint}.
+   *
+   * @param value data point value
+   */
+  public void setValue(Number value) {
+    this.value = value;
+  }
+
+  /**
+   * Compares the data point to another data point.
+   *
+   * @param p data point to compare
+   * @return true if the timestamp, type, and value are the same
+   */
+  public boolean equals(DataPoint p) {
+    boolean returnVal = false;
+    if (p instanceof DataPointNumber) {
+      returnVal =
+          p.getTimeStamp().equals(timestamp)
+              && (((DataPointNumber) p).getValue() == null
+                  ? value == null
+                  : ((DataPointNumber) p).getValue().equals(value))
+              && p.getTagName().equals(tagName);
+    }
+    return returnVal;
+  }
+
+  /**
+   * Compares the data point's value to another data point's value.
+   *
+   * @param p data point to compare
```

Comment:
Should change param to DataPointNumber. Then compiler will always check type. 
Disregard comment above